pkg/controller/perconaservermongodb/version.go

```diff
@@ -79,60 +101,103 @@ func (r *ReconcilePerconaServerMongoDB) ensureVersion(cr *api.PerconaServerMongo
 		return errors.New("cluster is not ready")
 	}
 
-	new := vs.CheckNew()
+	vm := versionMeta{
+		Apply:         string(cr.Spec.UpgradeOptions.Apply),
+		KubeVersion:   r.serverVersion.Info.GitVersion,
+		MongoVersion:  cr.Status.MongoVersion,
+		PMMVersion:    cr.Status.PMMVersion,
+		BackupVersion: cr.Status.BackupVersion,
+		CRUID:         string(cr.GetUID()),
+	}
+	if cr.Spec.Platform != nil {
+		vm.Platform = string(*cr.Spec.Platform)
+	}
 
-	if cr.Status.MongoVersion != new.MongoVersion {
-		log.Info(fmt.Sprintf("update Mongo version to %v", new.MongoVersion))
-		cr.Spec.Image = new.Image
-		cr.Status.MongoVersion = new.MongoVersion
+	newVersion, err := vs.GetExactVersion(vm)
+	if err != nil {
+		return fmt.Errorf("failed to check version: %v", err)
 	}
-	if cr.Status.BackupVersion != new.BackupVersion {
-		log.Info(fmt.Sprintf("update Backup version to %v", new.BackupVersion))
-		cr.Spec.Backup.Image = new.BackupImage
-		cr.Status.BackupVersion = new.BackupVersion
+
+	if cr.Status.MongoVersion != newVersion.MongoVersion {
```

Comment:
can we please compare Images? to make sure that we always use the right image which was returned by version service

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/385/commits/7faca94be7394e5b7de0b9afb6461eca76d6f6d6